### DISCUSSION TITLE
Tuning and general questions

### DISCUSSION THREAD

**Discussion by groot911** [bancorprotocol/fastlane-bot#56] [Q&A] [answered]
Hello y'all
I notice only a few wallets are managing to make all the profits, trying to optimize and fine tune mine, so please if anyone can shed light on the following:

1- is the  DEFAULT_GAS = 950_000 here in gwei? 
3- What about the default min profit which is 80? is that in BNT? i.e. if BNT is 0.5$ does that mean min profit is 40 in USD? and is that before of after splitting it half way with the Bancor fee burner? 
2- if i pas the configuration option --flashloan_tokens=WETH-6Cc2, does it optimize any calculations while using b3_two_hop  and would it only perform arbs for eth?
3- any best practice advice to optimize in the case of average internet speed?
4- why does "Data validation failed. Updating pools and restarting." occur? is it normal for that to happen IMMEDIATELY after finding arb opportunities i.e. 1 ms after?

Thanks!

**Comment by mikewcasale** [comment, marked as answer]
Hi, to the best of my knowledge, here are the answers to your questions:

1. Yes, `DEFAULT_GAS` is specified in gwei. However the `DEFAULT_GAS` value is only used for testing as far as I can tell. The actual gas submitted with transactions is estimated in real-time using web3.py methods.

2. The `--flashloan_tokens` flag refers to those token denominations which the bot can take a flash loan in. By default, these are `[WETH, DAI, USDC, USDT, WBTC, BNT, NATIVE_ETH]`. If you override the default to `TKN`, the search space is decreased for all modes, including the `b3_two_hop` mode (assuming that `--limit_bancor3_flashloan_tokens=True`).

3. The default minimum profit of 80 is in BNT.  The expected profit for a given arbitrage route is compared against the default minimum profit **before** the expected profit is adjusted by the `ARB_REWARD_PERCENTAGE`, which is read from the arb contract in real-time (currently this is 50%). Thus, if the expected profit for a given arbitrage route is greater than the minimum profit in BNT, then the _adjusted profit_ is calculated by multiplying times `ARB_REWARD_PERCENTAGE`, and if this value is greater than the estimated gas, the transaction will be attempted, otherwise it will be discarded.

4. In terms of optimizing for average internet speed, you might want to consider running the bot on a VPS with a high-speed connection located close to the Ethereum node you are using. This can reduce latency. 

5. "Data validation failed. Updating pools and restarting." This message appears when the bot attempts to validate data it has received via emitted contract events versus data it has read directly from the the contracts. The core data retrieval is done via events, however, due to some unavoidable timing issues, there are some cases when the most recently fetched event data is not current. In other words, the process is: _get-latest-events_-->_find-all-arbitrage-opportunities_-->_select-best-ops_-->_validate_-->_submit-tx_. In all likelihood, these steps require more than 1ms to complete, but regardless, **_any_** amount of time lapse after the get-latest-events step can theoretically invalidate the data.

Hope this helps!
